.github/tests/test_script.py

```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/python
+
+# Copyright (C) 2020 Vanessa Sochat.
+
+# This Source Code Form is subject to the terms of the
+# Mozilla Public License, v. 2.0. If a copy of the MPL was not distributed
+# with this file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from jsonschema import validate
+from jsonschema.exceptions import ValidationError
+
+import json
+import os
+import re
+import pytest
+import yaml
+
+here = os.path.dirname(os.path.abspath(__file__))
+root = os.path.dirname(os.path.dirname(here))
+
+
+def load_schema(path):
+    """load a schema from file. We assume a json file
+    """
+    with open(path, "r") as fd:
+        schema = json.loads(fd.read())
+    return schema
+
+
+def load_recipe(path):
+    """load a yaml recipe file
+    """
+    with open(path, "r") as fd:
+        content = yaml.load(fd.read(), Loader=yaml.SafeLoader)
+    return content
+
+
+def test_script_schema(tmp_path):
+    """the script test_organization is responsible for all the schemas
+       in the root of the repository, under <schema>/examples.
+       A schema specific test is intended to run tests that
+       are specific to a schema. In this case, this is the "script"
+       folder. Invalid examples should be under ./invalid/script.
+    """
+    print("Root of testing is %s" % root)
+
+    schema_name = "script"
+    schema_dir = os.path.abspath(os.path.join(root, schema_name))
+    print("Testing schema %s" % schema_name)
+
+    schemas = os.listdir(schema_dir)
+    for schema in schemas:
+        if schema.endswith("json"):
+
+            # Assert it loads with jsonschema
+            schema_file = os.path.join(schema_dir, schema)
+            loaded = load_schema(schema_file)
+
+            # Assert is named corretly
+            print("Getting version of %s" % schema)
+            match = re.search(
+                "%s-v(?P<version>[0-9]{1}[.][0-9]{1}[.][0-9]{1})[.]schema[.]json"
+                % schema_name,
+                schema,
+            )
+            assert match
+
+            # Ensure we found a version
+            assert match.groups()
+            version = match["version"]
+
+            # Ensure a version folder exists with invalids
+            print("Checking that invalids exist for %s" % schema)
+            invalids = os.path.join(here, "invalid", schema_name, version)
+            assert os.path.exists(invalids)
+            recipes = os.listdir(invalids)
+            assert recipes
```

Comment:
These two ``assert`` test will be shared across all schema since we require invalid directory. Makes sense to have a method for this that can be imported. Again something to refactor later.

Reply:
Sure thing.